### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by Florian95**
## Implementation of [Postman test API](https://github.com/gothinkster/realworld/tree/master/api) & JWT check

Method | Path | Status
------------ | ------------ | -------------
`POST` | `api/users` | ✅ 
`POST` | `/api/users/login` | ✅ 
`GET` | `/api/user` with JWT check | ✅ 
`PUT` | `/api/user` with JWT check | ✅ 
`GET` | `/api/profiles/rick` without JWT check | ✅ 
`GET` | `/api/profiles/rick` with JWT check | ✅ 
`POST` | `/api/profiles/rick/follow` with JWT check | ✅ 
`DELETE` | `/api/profiles/rick/unfollow` with JWT check | ✅ 

Phoenix tests are OK too.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,4 +6,4 @@ elixir:
 env:
   - MIX_ENV=test
 
-script: mix coveralls.travis
+script: mix test
```
Comment (BrainMaestro) [inline .travis.yml]: Would this report the coverage to coveralls?

Comment (Florian95) [inline .travis.yml]: No, but it's not configured on my repo, that's why I prefere to launch only the tests.

Comment (BrainMaestro) [inline .travis.yml]: Okay, but if this gets merged, we will lose coverage reports (even though they're sort of broken right now). Maybe you can keep the change just for yourself?

Comment (Florian95) [inline .travis.yml]: 👍

Comment (Florian95) [inline .travis.yml]: Done ! :)

**Comment by BrainMaestro** [comment]
Great! This looks excellent. Thanks for taking the time. Will merge soon.

**Comment by BrainMaestro** [comment]
Hey @Florian95. Looks like the work @fcevado is doing in #15 would affect this right? Since we want to change the context to just `Blog` and `Auth`. That means all the user related work you've done would fit into the `Auth` context instead of the current `User` one. That PR should be done soon.

**Comment by Florian95** [comment]
@BrainMaestro : His branch does exactly the same thing as mine.

But in the meantime, the tests are correct, the endpoints are fully compatible with the Postman Original project and the JWT layer is integrated.

Why, if you wish, do not take my branch and just rename User in Auth?

**Comment by BrainMaestro** [comment]
Okay since yours is more complete, can you rename it so I can merge. I don't mean to disregard what you've done. Sorrry if it came across that way. It's excellent

**Comment by Florian95** [comment]
Like this ? And I take some of the @fcevado changes.

**Comment by Florian95** [comment]
I use https://github.com/hexpm/hexpm/tree/master/lib/hexpm/accounts to refacto my branch. To be maybe more "Elixir" compliant.

Any feedbacks ?

**Comment by BrainMaestro** [comment]
This looks great.

image

It's currently failing the `token` test. It seems weird to return the user token though, but if it's the spec then I guess we have to implement it. What do you think?

**Comment by Florian95** [comment]
Yep, it's weird to return it here. But I can.

**Comment by BrainMaestro** [comment]
Okay, can you add that?

**Comment by Florian95** [comment]
Here it is ! :)

**Comment by Florian95** [comment]
New commits with :

Path | Status
------------ | -------------
`GET /api/profiles/rick` without JWT check | ✅ 
`GET /api/profiles/rick` with JWT check | ✅ 
`GET /api/profiles/rick/follow` with JWT check | ✅ 
`GET /api/profiles/rick/unfollow` with JWT check | ✅

**Comment by BrainMaestro** [comment]
Hey @Florian95. This is great, but could you separate the profile functionality from this. So that it could be a separate PR. It's easier to follow when it's one major feature per PR. I appreciate the effort though

**Comment by Florian95** [comment]
Profiles really depends on the previous work because it use all the JWT/Gardian/Users Context stuff. It's already "isolated" in a single commit : https://github.com/BrainMaestro/elixir-phoenix-realworld/pull/17/commits/3b0d2aee040f9cfff33246c90f2e366723ba1950
It's "only" 1 controller, 3 routes, 1 schema & two context methods.

Can you review it from here ? Sorry, I'll be more atomic on my next PR :(

**Comment by BrainMaestro** [comment]
I was just about to add that. 

So I'm about to merge, but I would like to clean up the history, so I'm going to rebase this to two different commits. One for adding the auth feature and another for the profiles. That's fine right?

**Comment by Florian95** [comment]
OK for me !

**Comment by BrainMaestro** [comment]
I know this took a while, but it was better for it. Thanks!

**Pull request merged**